ST_grid.c

```diff
@@ -1,686 +1,370 @@
 
-					if (year > 1 && UseSeedDispersal)
-						_set_sd_lyppt(i, j);
+					rgroup_Establish(); 		// Establish individuals. Excludes annuals.
 
+					Env_Generate();				// Generated the SOILWAT environment
 
+					rgroup_PartResources();		// Distribute resources
+					rgroup_Grow(); 				// Grow
 
-					rgroup_Establish(); /* excludes annuals */
+					mort_Main(&killedany); 		// Mortality that occurs during the growing season
 
-					Env_Generate(); //calls : SXW_Run_SOILWAT() which calls : _sxw_sw_run() which calls : SW_CTL_run_current_year()
+					rgroup_IncrAges(); 			// Increment ages of all plants
 
-					rgroup_PartResources();
-					rgroup_Grow();
+					grazing_EndOfYear(); 		// Livestock grazing
+					
+					save_annual_species_relsize(); // Save annuals before we kill them
 
-					mort_Main(&killedany);
+					mort_EndOfYear(); 			// End of year mortality.
 
-					rgroup_IncrAges();
+					stat_Collect(year); 		// Update the accumulators
 
-					// Added functions for Grazing and mort_end_year as proportional killing effect before exporting biomass end of the year
-					_do_grid_grazing_EndOfYear(i, j);
-					_do_grid_disturbances(i, j);
+				    _kill_annuals(); 			// Kill annuals
+				    _kill_maxage();             // Kill plants that reach max age
+					proportion_Recovery(); 		// Recover from any disturbances
+					_kill_extra_growth(); 		// Kill superfluous growth
 
-					stat_Collect(year);
-
-					_save_cell(i, j, year, TRUE);
-
-		            // Moved kill annual and kill extra growth after we export biomass, we also doing recoverly after killing year
-				    _kill_annuals();
-					_do_grid_proportion_Recovery(i, j);
-					_kill_extra_growth();
-
-					if (UseProgressBar)
-					{
-						prog_Percent += prog_Incr; //updating our percent done
-						if (prog_Percent > prog_Acc)
-						{ //only update if 1% progress or more has been made since the last time we updated (this check is so it doesn't waste processing time that could be spent running the simulations by updating all of the time)
-							prog_Acc += 0.01;
-							_load_bar(prog_Prefix, prog_Time,
-									(int) (100 * prog_Percent), 100, 100, 10); //display our bar to the console
-						}
-					}
 				} /* end model run for this cell*/
-
-			if (UseSeedDispersal)
-				_do_seed_dispersal();
+			} /* end model run for this row */
+			if (UseSeedDispersal){
+				disperseSeeds();
+            }
+			
+			unload_cell(); // Reset the global variables
 		}/* end model run for this year*/
 
-		// collects the data appropriately for the mort output... (ie. fills the accumulators in ST_stats.c with the values that they need)
-		if (MortFlags.summary)
-			for (i = 1; i <= grid_Rows; i++)
-				for (j = 1; j <= grid_Cols; j++)
+		// collects the data for the mort output,
+        // i.e. fills the accumulators in ST_stats.c.
+		if (MortFlags.summary){
+			for (i = 0; i < grid_Rows; i++){
+				for (j = 0; j < grid_Cols; j++)
```

Comment:
@kpalmqui correct. I changed the 1-indexing to 0-indexing because all C arrays are 0-indexed, so a C programer should expect to begin iterations at 0.